example/eblob_backend.cpp

```diff
@@ -42,6 +42,53 @@
 #include "rapidjson/writer.h"
 #include "rapidjson/stringbuffer.h"
 
+
+constexpr auto bugfix_timestamp = EBLOB_BUGFIX_TIMESTAMP;
+constexpr auto stamp_headers_size = sizeof(eblob_disk_control) + sizeof(dnet_ext_list_hdr);
+
+
+// Checks for broken headers signature(s): ELL-817
+int blob_check_corrupted_stamp(const eblob_disk_control &dc, const dnet_ext_list_hdr &ehdr) {
+	const auto check_dc_header = [] (const eblob_disk_control &dc) {
+		constexpr auto flags_limit = 1 << 9;
+		return dc.flags && dc.flags < flags_limit;
+	};
+
+	const auto check_ext_header = [] (const dnet_ext_list_hdr &ehdr) {
+		return ehdr.version == DNET_EXT_VERSION_V1 &&
+		       ehdr.size &&
+		       ehdr.timestamp.tsec <= bugfix_timestamp &&
+		       !(ehdr.__pad1[0] || ehdr.__pad1[1] || ehdr.__pad1[2]) &&
+		       !(ehdr.__pad2[0] || ehdr.__pad2[1]);
+	};
+
+	if (check_dc_header(dc) && check_ext_header(ehdr)) {
+		return -EILSEQ;
+	}
+
+	return 0;
+}
+
+static inline int read_and_check_stamp(int fd, uint64_t data_size, uint64_t data_offset) {
+    using stamp_array = std::array<char, stamp_headers_size>;
+
+    stamp_array stamp;
+
+    if (data_size < stamp_headers_size) {
```

Comment:
You can use `EBLOB_BUGFIX_TIMESTAMP` in array declaration and `stamp.size()` here.